thunder/core/trace.py

```diff
@@ -358,15 +358,17 @@ def keyfn(class_or_module: type | ModuleType) -> str:
                         import_str = f"import {module.__name__} as {name}"
                 program.append(import_str)
 
-            program.append("from thunder.executors.torchex import no_autocast")
             # Separates imports from the function for readability
             if len(import_ctx) > 0:
                 program.append("")
 
-            # Prints the signature and the no_grad context (for when calling torch operations)
+            # Disable gradients since Thunder takes care of this (for when calling torch operations)
             program.append("@torch.no_grad()")
-            # Prints the signature and the no_autocast context
-            program.append("@no_autocast()")
+            # Disable autocast since we already generated the trace with it in consideration (for when calling torch
+            # operations)
+            program.append('@torch.autocast(device_type="cuda", enabled=False, cache_enabled=False)')
+            program.append('@torch.autocast(device_type="cpu", enabled=False, cache_enabled=False)')
+            # Prints the signature
```

Comment:
Is it possible to register no_autocast in Dynamo instead?

Reply:
I'm not familiar with registration mechanisms in dynamo. It might be possible.

However, what's the disadvantage of calling the torch symbol? We already have the precedent of `torch.no_grad()`